targets/csharp-unityv2/source/Shared/Internal/PlayFabHttp/PlayFabWebRequest.cs

```diff
@@ -0,0 +1,596 @@
+#elif UNITY_IOS
+                                    attribRequest.Idfa = advertisingId;
+#endif
+                                        PlayFabClientAPI.AttributeInstall(attribRequest, (attribResult) =>
+                                {
+                                //This is for internal testing tools.
+                                PlayFabSettings.AdvertisingIdType += "_Successful";
+                                }, null);
+                                    }
+                                });
+                        });
+                    }
+#endif
+#endif
+
+                    #endregion
+                }
+
+                var cloudScriptUrl = result as GetCloudScriptUrlResult;
+                if (cloudScriptUrl != null)
+                {
+                    PlayFabSettings.LogicServerUrl = cloudScriptUrl.Url;
+                }
+#endif
+                //Send our post event.  //TODO: maybe we want this Enqueued also?
+                PlayFabHttp.SendEvent(request, result, ApiProcessingEventType.Post);
+
+                if (callBack != null)
+                {
+                    lock (ResultQueue)
+                    {
+                        //Queue The result callbacks to run on the main thread.
+                        ResultQueue.Enqueue(() => {
+#if PLAYFAB_REQUEST_TIMING
+                            state.Stopwatch.Stop();
+                            state.Timing.MainThreadRequestMs = (int)state.Stopwatch.ElapsedMilliseconds;
+                            PlayFabHttp.SendRequestTiming(state.Timing);
+#endif
+                            callBack(result);
+                        });
+                    }
+                }
+
+            }
+            catch (Exception e)
+            {
+                Debug.LogException(e);
+                state.State = HttpRequestState.Error;
+            }
+
+        }
+
+        
+        public void Update()
+        {
+            lock (ResultQueue)
+            {
+                while (ResultQueue.Count > 0)
+                {
+                    var actionToQueue = ResultQueue.Dequeue();
+                    _tempActions.Enqueue(actionToQueue);
+                }
+            }
+
+            while (_tempActions.Count > 0)
+            {
+                var finishedRequest = _tempActions.Dequeue();
+                finishedRequest.Invoke();
+            }
+        }
+
+        public static string ResponseToString(HttpWebResponse webResponse)
+        {
+            try
+            {
+                using (var responseStream = webResponse.GetResponseStream())
+                {
+                    if (responseStream == null)
+                        return null;
+                    using (var stream = new System.IO.StreamReader(responseStream))
+                    {
+                        return stream.ReadToEnd();
+                    }
+                }
+
+            }
+            catch (WebException e)
```

Comment:
You should only process WebException differently than Exception is if there's different/extra data in WebException.  Which there is.  And you're not fetching it.  Go back and look at how V1 processes WebException.